modules/core/04-channel/keeper/upgrade.go

```diff
@@ -85,6 +88,36 @@ func (k Keeper) WriteUpgradeTryChannel(
 	// grab channel inside this function to get most current channel status
 }
 
+// validateUpgradeFields validates the proposed upgrade fields against the existing channel.
+// It returns an error if the following constraints are not met:
+// - there exists at least one valid proposed change to the existing channel fields
+// - the proposed order is a subset of the existing order
+// - the proposed connection hops do not exist
+// - the proposed version is non-empty (checked in UpgradeFields.ValidateBasic())
```

Comment:
Is this comment still true for this function?

Reply:
Yes it is checked in `MsgChannelUpgradeInit.ValidateBasic()` which calls `msg.Fields.ValidateBasic()`, would you like me to update the comment to be more specific?